server/src/test/java/org/kaazing/gateway/server/preferedipv4/ResourceAddressFactorySpiPreferIPV4Test.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright 2007-2015, Kaazing Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kaazing.gateway.server.preferedipv4;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.net.URI;
+
+import org.junit.Test;
+import org.kaazing.gateway.server.test.Gateway;
+import org.kaazing.gateway.server.test.config.GatewayConfiguration;
+import org.kaazing.gateway.server.test.config.builder.GatewayConfigurationBuilder;
+
+/**
+ * Class for testing preferIPv4Stack behavior
+ */
+public class ResourceAddressFactorySpiPreferIPV4Test {
+
+    private static final String JAVA_NET_PREFER_IPV4_STACK = "java.net.preferIPv4Stack";
+
+    @Test
+    public void testEmptyCollectionWithFlagTrue() {
```

Comment:
No need for test prefix in junit for the methods, as they are identified by class name#test so ResourceAddressFactorySpiPreferIPV4Test#testEmptyCollectionWithFlagTrue is redundant.  Naming is hard, I would suggest ResourceAddressFactorySpiPreferIPV4Test#failsStartupWhenMatchingIPv6AddressesNotFound, though you can probably come up with something better